enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,160 @@
+---
+title: csi-proxy
+authors:
+  - "@sebsoto"
+reviewers:
+  - "@openshift/openshift-team-windows-containers"
+  - "@openshift/openshift-team-storage", for potential pitfalls that could be ran into.
+approvers:
+  - "@aravindhp"
+creation-date: 2023-03-29
+last-updated: 2023-03-29
+tracking-link:
+  - https://issues.redhat.com/browse/OCPBU-465
+---
+
+# Enable running CSI drivers on Windows Nodes through CSI Proxy
+
+## Release Sign-off Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+This goal of this enhancement is to enable the use persistent storage for Windows workloads, without the use of in-tree
+storage drivers, [which are in the process of being deprecated and removed.](https://kubernetes.io/blog/2022/09/26/storage-in-tree-to-csi-migration-status-update-1.25/)
+
+This will be done by running [CSI Proxy](https://github.com/kubernetes-csi/csi-proxy) as a Windows service on each
+Windows node. csi-proxy is a pre-requisite for CSI node driver pods to interact with the underlying Windows storage
+APIs. With csi-proxy running on a cluster's Windows nodes, cluster administrators are free to deploy the CSI storage
+drivers of their choice as a Windows DaemonSet, enabling Windows pods on each node to have access to storage through
+PersistentVolumeClaims.
+
+## Motivation
+
+Windows nodes on OpenShift have had persistent storage available through the use of the in-tree storage drivers.
+Persistent storage is a necessary feature for production workloads on Kubernetes, and it is important that this
+functionality continue to work, even with the in-tree drivers being removed.
+
+### User Stories
+
+* As a developer using OpenShift, I want the Windows workloads I create to be able to share storage between workloads,
+  and have that storage persist when pods are removed and re-created.
+
+### Goals
```

Comment:
It should be explicitly called out somewhere if using the volume by Linux and Windows Pods is a requirement or not. It might be tricky due to filesystem support, not sure if / how we support NTFS on Linux or ext4/xfs on Windows.

Reply:
@jsafrane this has been called out in non-goals now. Does that suffice?

Reply:
Yes